src/test/test_connector_redis.cpp

```diff
@@ -0,0 +1,348 @@
+/*************************************************************************
+ * libjson-rpc-cpp
+ *************************************************************************
+ * @file    test_connector_redis.cpp
+ * @date    13.08.2017
+ * @author  Jacques Software <software@jacques.com.au>
+ * @license See attached LICENSE.txt
+ ************************************************************************/
+
+#ifdef REDIS_TESTING
+#include <catch.hpp>
+#include <jsonrpccpp/server/connectors/redisserver.h>
+#include <jsonrpccpp/client/connectors/redisclient.h>
+
+#include "mockclientconnectionhandler.h"
+#include "checkexception.h"
+
+#include <unistd.h>
+#include <sstream>
+#include <iostream>
+#include <signal.h>
+
+using namespace jsonrpc;
+using namespace std;
+
+#define TEST_PORT 6379
+#define TEST_HOST "127.0.0.1"
+#define TEST_QUEUE "mytest"
+
+#define TEST_MODULE "[connector_redis]"
+
+#define REDIS_BIN "redis-server"
+#define REDIS_BIN_DIR "/usr/bin/"
+#define REDIS_CONF "redis.conf"
+
+
+
+namespace testredisserver
+{
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Start Redis server in the background
+    //
+
+    /**
+     * This method will kill the redis server when we are done with it.
+     * @param pid The pid of the redis server to kill.
+     */
+    void stop_redis_server(int pid) {
```

Comment:
Could you move the initialisation of the redis binary to a separate class? e.g. `TestRedisServer`